core/src/main/java/org/apache/camel/kafkaconnector/CamelSinkTask.java

```diff
@@ -175,8 +180,8 @@ public void put(Collection<SinkRecord> sinkRecords) {
             TaskHelper.logRecordContent(LOG, loggingLevel, record);
 
             Exchange exchange = new DefaultExchange(producer.getCamelContext());
-            exchange.getMessage().setBody(record.value());
-            exchange.getMessage().setHeader(KAFKA_RECORD_KEY_HEADER, record.key());
+            exchange.getMessage().setBody(convertValueFromStruct(record.valueSchema(), record.value()));
```

Comment:
Is this safe for NPEs on NULL-records which can arrive from SMTs under certain conditions?